doc/man3/EVP_DigestInit.pod

```diff
@@ -182,7 +182,12 @@ EVP_MD_meth_set_app_datasize().
 
 =item EVP_MD_CTX_md()
 
-Returns the B<EVP_MD> structure corresponding to the passed B<EVP_MD_CTX>.
+Returns the B<EVP_MD> structure corresponding to the passed B<EVP_MD_CTX>. This
+will be the same B<EVP_MD> object originally passed to EVP_DigestInit_ex() (or
+other similar function) when the EVP_MD_CTX was first initialised. Note that
+where explicit fetch is in use (see L<EVP_MD_fetch(3)) the value returned from
+this function will not have its reference count incremented and therefore it
+should not be used after the EVP_MD_CTX is freed.
```

Comment:
So errr....  what happens if the passed `EVP_MD` was a fetched one?

Reply:
You mean if it is an implicitly fetched one? Then... "This will be the same B<EVP_MD> object originally passed to EVP_DigestInit_ex()"

Reply:
No, sorry, I meant an explicitly fetched one.  Should that one be reusable?

Reply:
If its an explicitly fetched on then ...."This will be the same B<EVP_MD> object originally passed to EVP_DigestInit_ex()" and then the following note applies "Note that where explicit fetch is in use (see L<EVP_MD_fetch(3)) the value returned from this function will not have its reference count incremented and therefore it should not be used after the EVP_MD_CTX is freed."

I think this is acceptable because explicit fetch is new so we get to define how it works. We *could* have this function always return a const EVP_MD object. But that, I think, would be quite surprising if you didn't pass it one of those in the first place.

Reply:
Ok.  Thanks for illuminating the text for me.